### PR TITLE
Added a command line option to output a single hour

### LINKED ISSUE #3: Output a single hour
Through an argument? (`--single`)

### PR DESCRIPTION
**Pull request by Badboy-16**
I tried to fix #3 

I added the `-s` (`--single`) option to the command line input and enabled the programme to output a single hour.

Usage:
Output a single hour for current time.
```
$ timezone-converter hong_kong tokyo -s now
$ timezone-converter hong_kong tokyo --single now
```

Output a given hour specified by the user (e.g. 16).
```
$ timezone-converter hong_kong tokyo -s 16
$ timezone-converter hong_kong tokyo --single 16
```

### CODE REVIEW

File: timezone_converter/comparison_view.py [hunk 1]
```diff
@@ -70,6 +77,6 @@ def _build_table(self) -> Table:
 
         return table
 
-    def print_table(self) -> int:
-        self._print_with_rich(self._build_table())
+    def print_table(self, single_hour) -> int:
+        self._print_with_rich(self._build_table(single_hour))
```
Comment (ibLeDy) [inline timezone_converter/comparison_view.py:81]: As suggested on `main.py`, we would use it as instance argument, so no need to pass it here.

File: timezone_converter/comparison_view.py [hunk 2]
```diff
@@ -55,7 +55,14 @@ def _build_table(self) -> Table:
         fmt = '%Y-%m-%d %H:%M'
 
         current_hour = datetime.now().hour
-        for hour in range(24):
+        if single_hour is None:
+            hour_range = range(24)
+        elif single_hour == 'now':
+            hour_range = range(current_hour, current_hour + 1)
+        else:
+            hour_range = range(int(single_hour), int(single_hour) + 1)
```
Comment (ibLeDy) [inline timezone_converter/comparison_view.py:63]: I prefer to use something like we discussed in #15, a default value that will later change. I also prefer that the argument is just called `hour` (and that we provide it as an instance argument, as I mentioned in `main.py`). This is how this part would end up looking like:

```py
        hour_range = range(24)
        if self.hour is not None:
            hour_range = range(self.hour, self.hour + 1)
```

As I mentioned before, we would not use `now` and the current hour will be provided by `const` when no value is provided to `--single`.

We are also converting the value to `int` so no need to do it here.

File: timezone_converter/comparison_view.py [hunk 3]
```diff
@@ -55,7 +55,14 @@ def _build_table(self) -> Table:
...
+
+        for hour in hour_range:
```
Comment (ibLeDy) [inline timezone_converter/comparison_view.py:65]: As `current_hour` would only be used on this for loop, I would prefer that we move the `current_hour` declaration to the line before this one (line 64)

File: timezone_converter/comparison_view.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def _get_headers(self) -> List[str]:
 
         return headers
 
-    def _build_table(self) -> Table:
+    def _build_table(self, single_hour) -> Table:
```
Comment (ibLeDy) [inline timezone_converter/comparison_view.py:49]: As suggested on main.py, we would use it as instance argument, so no need to pass it here.

**Review by ibLeDy — CHANGES_REQUESTED** [review]

**Comment by Badboy-16** [comment]
Just pushed some commits based on your comments above. I think it's definitely a good idea to use a default value that can be changed later just like in #15 . Thanks for the reminder to utilise it again! 👍🏼 

I tested the output with the single hour option and it looks working fine.

**Comment by ibLeDy** [comment]
pre-commit.ci run

**Comment by ibLeDy** [comment]
We will have to wait a bit for `pre-commit` to run, because [seems like GitHub is having issues](https://www.githubstatus.com/):

Screenshot from 2021-05-16 16-12-35

**Comment by ibLeDy** [comment]
pre-commit.ci run

**Comment by ibLeDy** [comment]
pre-commit.ci autofix

**Comment by ibLeDy** [comment]
This looks good, I will create a release in a bit, thanks!

**Pull request merged**